appendix/stata-guide.tex

```diff
@@ -153,151 +198,241 @@ \subsection{Abbreviating commands}
 
 \codeexample{stata-boilerplate.do}{./code/stata-boilerplate.do}
 
 \subsection{Saving data}
 
-Similarly to boilerplate code, there are good practices that should be followed before saving the data set.
-These are sorting and ordering the data set, dropping intermediate variables that are not needed, and
-compressing the data set to save disk space and network bandwidth.
-
-If there is an ID variable or a set of ID variables, then the code should also test that they are uniqueally and
-fully identifying the data set.\sidenote{\url{https://dimewiki.worldbank.org/wiki/ID\_Variable\_Properties}} ID
-variables are also perfect variables to sort on, and to order leftmost in the data set.
-
-The command \texttt{compress} makes the data set smaller in terms of memory usage without ever losing any
-information. It optimizes the storage types for all variables and therefore makes it smaller on your computer
+There are good practices that should be followed before saving any data set.
+These are to \texttt{sort} and \texttt{order} the data set,
+dropping intermediate variables that are not needed,
+and compressing the data set to save disk space and network bandwidth.
+
+If there is a unique ID variable or a set of ID variables,
+the code should test that they are uniqueally and
+fully identifying the data set.\sidenote{
+  \url{https://dimewiki.worldbank.org/wiki/ID\_Variable\_Properties}}
+ID variables are also perfect variables to sort on,
+and to \texttt{order} first in the data set.
+
+The command \texttt{compress} makes the data set smaller in terms of memory usage
+without ever losing any information.
+It optimizes the storage types for all variables
+and therefore makes it smaller on your computer
 and faster to send over a network or the internet.
 
 \codeexample{stata-before-saving.do}{./code/stata-before-saving.do}
 
+\subsection{Miscellaneous notes}
+
+Use wildcards in variable names (\texttt{xx\_?\_*\_xx}) sparingly,
+as they may change results if the dataset changes.
+Write multiple graphs as \texttt{tw (xx)(xx)(xx)}, not \texttt{tw xx||xx||xx}.
+Put spaces around each binary operator except \texttt{\^}.
+Therefore write \texttt{gen z = x + y} and \texttt{x\^}\texttt{2}.
+When order of operations applies, use spacing and parentheses:
+\texttt{hours + (minutes/60) + (seconds/3600)}, not \texttt{hours + minutes / 60 + seconds / 3600}.
+For long expressions, the operator starts the new line, so:
+
+\texttt{gen sumvar =   x ///}
+
+\texttt{             + y ///}
+
+\texttt{             - z ///}
+
+\texttt{             + a*(b-c)}
+
+\noindent Make sure your code doesn't print very much to the results window as this is slow.
+This can be accomplished by using \texttt{run file.do} rather than \texttt{do file.do}.
+Run outputs like \texttt{reg} using the \texttt{qui} prefix.
+Never use interactive commands like \texttt{sum} or \texttt{tab} in dofiles,
+unless they are combined with \texttt{qui} for the purpose of getting \texttt{r()}-statistics.
```

Comment:
```suggestion
unless for the purpose of getting \texttt{r()}-statistics.
```
Why `qui`? I can agree taht is a good preference, but I do not see a reason to say that you cannot use `sum` and `tab` unless combined with `qui`

Reply:
I guess the point is more general -- which is, nothing you do should be intended to be read in the results window, so well-styled code doesn't slow down execution by printing things to the results window. Does that make more sense to write out explicitly?